src/clib/pioc_support.cpp

```diff
@@ -4762,7 +4752,7 @@ int PIOc_openfile_retry_impl(int iosysid, int *ncidp, int *iotype, const char *f
                 spio_ltimer_stop(ios->io_fstats->tot_timer_name);
                 spio_ltimer_stop(file->io_fstats->rd_timer_name);
                 spio_ltimer_stop(file->io_fstats->tot_timer_name);
-                return pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                return pio_err(ios, NULL, PIO_EADIOS2ERR, __FILE__, __LINE__,
```

Comment:
Looks like the file is still valid here, right? Why did you change file to NULL?

Reply:
The file description data has not been fully populated, so a NULL pointer should be used when calling pio_err().